lib/workers/branch/reuse.ts

```diff
@@ -49,7 +49,7 @@ export async function shouldReuseExistingBranch(
   if (
     config.rebaseWhen === 'behind-base-branch' ||
     (config.rebaseWhen === 'auto' && (await platform.getRepoForceRebase())) ||
-    (config.automerge && config.automergeType === 'branch')
+    (config.rebaseWhen !== 'never' && config.automerge === true)
```

Comment:
What about when `config.rebaseWhen === 'conflicted'`? If someone configures that plus automerge then they might not want it rebased before automerging?

Reply:
🤔 Are you sure? We can't detect that at that position. We are already behind (conflicted or not). Mostly it will conflicted because of another automerged pr (lockfile conflict) i think it should auto resolve by updating the branch

Reply:
We would ideally be able to support two scenarios:
- People who want **only** automerging PRs up-to-date (rebaseWhen=auto)
- People who want PRs automerged unless conflicted (rebaseWhen=conflicted)

Is it really not possible here?

Reply:
in this case we should move th conflicted pr check above the branch check, because we don't know if there is a conflict.

we only know we are behind, until we would try to merge current branch to target.